File: server/src/modules/user/user.model.js
const userModel = {
    table: "users",
    field: {
        userId: "user_id",
        fullName: "full_name",
        email: "email",
        phone: "phone",

        // Quyền ADMIN | VICTIM | RESCUER
        role: "role", 

        // ONLINE | OFFLINE
        status: "status",

        avatarUrl: "avatar_url",
        isVerified: "is_verified",

        // Cập nhật khi nhấn vị trí của tôi hoặc gửi request
        currentLat: "current_lat",
        currentLng: "current_lng",

        createdAt: "created_at",
        updatedAt: "updated_at"
    },
};

module.exports = userModel;

File: mobile/lib/feature/user/user_model.dart
class UserModel {
  final String userId;
  final String fullName;
  final String email;
  final String? phone;
  final String role;
  final String status;
  final String? avatarUrl;
  final bool isVerified;

  UserModel({
    required this.userId,
    required this.fullName,
    required this.email,
    this.phone,
    required this.role,
    required this.status,
    this.avatarUrl,
    required this.isVerified,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      userId: json['userId'] ?? '',
      fullName: json['fullName'] ?? '',
      email: json['email'] ?? '',
      phone: json['phone'], // nullable
      role: json['role'] ?? '',
      status: json['status'] ?? '',
      avatarUrl: json['avatarUrl'], // nullable
      isVerified: json['isVerified'] ?? false,
    );
  }
}